# Z-Test

In [1]:
import numpy as np
import scipy as sp

## 1 Introduction to Z-Test

### 1.1 Assumptions for Z-test

1. Population Standard Deviation $\sigma$ is known.
2. Z-test is used for testing hypothesis on population parameter type - Mean.
3. Sampling distribution of sample mean is normal and its mean is available.

### 1.2 Conditions for Z-test

1. Data should be of type Continuous.
2. Random Sampling.
3. Observations are independent of each other.
4. Sample size $n \ge 30$.
5. Known population standard deviation.

### 1.3 Application of CLT in Z-Test

#### How CLT applies to Hypothesis testing?

1. There is some population distribution, with mean $\mu$ and standard deviation $\sigma$ which may or may not be Gaussian (Normal distribution).
2. We take mean sampling distribution $\bar{X}$ of size $m$ (where each sample is of size $n$) from the population distribution.
3. **As per CLT** this mean sampling distribution will be a Normal distribution $\displaystyle \bar{X} \sim N\biggl(\mu\;, \frac{\sigma}{\sqrt{n}}\biggr)$.
4. Standardization of this $\bar{X}$ distribution creates **Standard Normal Distribution** $Z$.
5. This Standard Normal Distribution acts as **Reference Distribution** used in decision-making step for Hypothesis Testing.
6. We use this reference distribution $Z$ to calculate $\text{Z-Score}$.
7. Using $\text{Z-Score}$, calculate area under the curve, called as $\text{p-value}$, used in decision-making step of HT.

> **Note**:
>
> Mean sampling distribution $\bar{X}$ is defined as sample distribution of sample means.

### 1.4 Critical Value

#### Definition

Minimum value - $\text{x-critical}$ - at which the required confidence level is reached.

#### Properties

At Z-Critical, $\text{p-value}$ is equal to Significance level $\alpha$.

#### Steps to calculate Critical value

- **STEP 1**: Use `ppf` function to get $\text{z-critical}$ value from a standard normal distribution.
- **STEP 2**: Convert $\text{z-critical}$ value to critical value (of a random variable $X$) using linear transformation.

#### Linear Transformation

$$
\large
\text{x-critical} = \mu + \bigl(\text{Z-critical} \cdot \sigma\bigr)
$$

Where,

- $\text{x-critical}$ - Critical value
- $\text{Z-critical}$ - Special $\text{Z-score}$ value
- $\sigma$ - Standard Deviation

#### Critical Region

![Critical Region](images/critical_region.png)

- Region beyond critical-value is called as Critical region.
- Reject $H_0$ if $\text{p-value}$ falls in this region.

#### Difference Between x-critical, Z-critical and Alpha

- Significance level ($\alpha$) is percentage value like 0.05 or 5% etc.
- $\text{Z-critical}$ is a special value of Z-score test-statistic on x-axis.
- $\text{Z-critical}$ should undergo linear transformation to get the actual Critical value ($\text{x-critical}$).
- Use `ppf` function to convert Significance level to $\text{Z-critical}$.

$$
\large
\text{norm.ppf($1 - \alpha$)}\,\rightarrow\,\text{Z-critical}\,\rightarrow\,\text{x-critical}
$$

### 1.5 Application of CI in Z-Test

$$
\large
\text{Confidence Interval} = \text{Point Estimate} \pm \text{Margin of Error}
$$

#### How CI applies to Hypothesis testing?

Steps to calculate Confidence Interval:

1. Calculate confidence level using the significance level.
2. Calculate $\text{Z-critical}$ using confidence level.
3. Calculate margin of error (moe) using $\text{Z-critical}$ and Standard Error.
4. Add and subtract moe from sample mean to get Confidence Interval.

Steps to calculate Confidence Interval Given sample mean $\mu_{\text{sample}}$, standard-deviation $\sigma$ of a sample pf size $n$:

$$
\large
\begin{aligned}
\text{Point Estimate}      &= \mu_{\text{sample}} \\[10pt]
\text{Margin of Error}     &= \text{Z-critical} \cdot \frac{\sigma}{\sqrt{n}} \\[10pt]
\text{Confidence Interval} &= \mu_{\text{sample}} \pm \biggl(\text{Z-critical} \cdot \frac{\sigma}{\sqrt{n}} \biggr)
\end{aligned}
$$

#### Hypothesis Testing Framework using CI

- **STEP #1**: Setup Null-Hypothesis $H_0$ and Alternate-Hypothesis $H_a$
- **STEP #2**: Choose a Distribution, Test-statistic and Significance-level $(\alpha)$
- **STEP #3**: Select type of test: Left / Right / Two-tailed
- **STEP #4**: Compute Confidence-Interval.
- **STEP #5**: Failed to Reject $H_0$ if population parameter falls in between the Confidence Interval, otherwise reject $H_0$.

#### Does right/left tailed test have CI?

1. Yes right/left tailed tests can have confidence intervals.
2. Example CI for left tailed test: $[-\infty,\;\text{x-critical}]$
3. Example CI for right tailed test: $[\text{x-critical},\;\infty]$

> **Note**:
>
> Confidence Interval is plotted on a Sampling Distribution and not on a Reference distribution.

### 1.6 Types of Z-tests

There are two types of Z-test:

1. One Sample Z-test
2. Two sample Z-test

## 2 One Sample Z-test

### 2.1 Introduction

#### Why is it called One Sample Z-test?

In One Sample Z-test, mean of one single sample is compared against population mean.

#### When to use One Sample Z-test?

- When we have values of population parameters like mean and standard deviation.
- When we want to compare mean values between sample distribution and population distribution.

### 2.2 Nature of Hypotheses

- $H_0: \mu = \mu_0$
- $H_a: \mu \neq \mu_0\;(\,\text{or}\; \mu > \mu_0 \,,\, \mu < \mu_0\,)$

Where,

- $\mu$ - Variable representing Population Mean
- $\mu_0$ - Value of Population Mean

#### Examples

##### Example #1

- $H_0: \mu = 1800$
- $H_a: \mu \neq 1800$

##### Example #2

- $H_0: \mu = 1800$
- $H_a: \mu < 1800$

##### Example #3

- $H_0: \mu = 1800$
- $H_a: \mu > 1800$

### 2.3 Test Statistic

1. The name of the test statistic used in one sample Z-test is called as $\text{Z-Score}$.
2. Sample distribution of sample means follows Normal Distribution.

#### Reference Distribution

Reference Distribution is **Z Distribution** (Standard Normal Distribution) with mean $\mu = 0$ and standard deviation $\sigma = 1$.

#### Formula

Formula to calculate $\text{Z-Score}$:

$$
\large
\text{Z-Score} = \frac{x - \mu}{\frac{\sigma}{\sqrt{n}}}
$$

### 2.4 Conditions

1. Population parameters like $\mu$ and $\sigma$ are available.
2. Sample size $n$ is large enough. Typically $n \ge 30$.
3. Single sample's parameters are compared against population parameters.

## 3 Two Sample Z-test

### 3.1 Introduction

#### Why is it called Two Sample Z-test?

In Two Sample Z-test, mean of one sample is compared against mean of another sample.

#### When to use Two Sample Z-test?

- When we have values of population parameters like mean and standard deviation.
- When we have to compare mean values between two sample distributions from same or different population.

### 3.2 Nature of Hypotheses

Based on problem statement Hypotheses can can be defined as:

- $H_0: \mu_1 = \mu_2$
- $H_a: \mu_1 \neq \mu_2 \;(\,\text{or}\; \mu_1 > \mu_2 \,,\, \mu_1 < \mu_2\,)$

or,

- $H_0: \mu_1 - \mu_2 \simeq 0$
- $H_a: \mu_1 - \mu_2 \ne 0 \;(\,\text{or}\; \mu_1 - \mu_2 > 0 \,,\, \mu_1 - \mu_2 < 0\,)$

Where,

- $\mu_1$ - Mean value of Sample #1.
- $\mu_2$ - Mean value of Sample #2.

### 3.3 Test Statistic

1. The name of the test statistic used in two sample Z-test is called as Z-score.
2. Sample distribution of sample means follows Normal Distribution.

#### Reference Distribution

Reference Distribution is **Z Distribution** (Standard Normal Distribution) with mean $\mu = 0$ and standard deviation $\sigma = 1$.

#### Formula

##### Case #1

Formula to calculate Z-Score when both the samples are from same population:

$$
\large
\text{Z-Score} = \frac{\mu_1 - \mu_2}{\sqrt{\frac{\sigma_1^2}{n_1} + \frac{\sigma_2^2}{n_2}}}
$$

Where,

- $\mu_1$ - Mean value of Sample #1.
- $\mu_2$ - Mean value of Sample #2.
- $\sigma_1$ - Standard deviation value of Sample #1.
- $\sigma_2$ - Standard deviation value of Sample #2.

##### Case #2

Formula to calculate Z-Score when both the samples are from different population:

$$
\large
\text{Z-Score} = \frac{(\bar{x}_1 - \bar{x}_2) - (\mu_1 - \mu_2)}{\sqrt{\frac{\sigma_1^2}{n_1} + \frac{\sigma_2^2}{n_2}}}
$$

Where,

- $\mu_1$ - Mean value of Population distribution #1.
- $\mu_2$ - Mean value of Population distribution #2.
- $\bar{x}_1$ - Mean value of Sample distribution #1.
- $\bar{x}_2$ - Mean value of Sample distribution #2.
- $\sigma_1$ - Standard deviation value of Population distribution #1.
- $\sigma_2$ - Standard deviation value of Population distribution #2.

### 3.4 Conditions

1. Population parameters like $\mu$ and $\sigma$ are available.
2. Sample size $n$ is large enough. Typically $n \ge 30$.
3. Sample statistic of Sample-1 is compared against Sample static of Sample-2.
4. Two samples can be of different size but must be i.i.d.
5. Two samples can be from different population.

> **Note**:
>
> i.i.d := Independent and Identically Distributed.

## 4 Z-Test Template

### 4.1 Hypothesis Testing using p-value

#### STEP #1

Define Null-Hypothesis and Alternate-Hypothesis.

1. $H_0$ :
2. $H_a$ :

#### STEP #2

Identify:

1. Distribution:
2. Test Statistic: z-score
2. Significance level:

```python
alpha =   # Significance level.
```

#### STEP #3

Identify tail-type based on observed value.

```python
x =   # Observed value.
```

#### STEP #4

Compute $\text{p-value}$

```python
import numpy as np
import scipy as sp

# Given,
mu =      # Mean
sigma =   # Standard deviation
n =       # Sample size

se = sigma / np.sqrt(n)  # Standard error

# x = mu + (z * sigma)
# For sample distribution Standard error becomes sigma.
z = (x - mu) / se

# Compute p_value using `stats.norm.cdf(z)`
# Left tailed test.
p_value = sp.stats.norm.cdf(z)
# or
# Right tailed test.
p_value = 1 - sp.stats.norm.cdf(z)
# or
# Two tailed test.
p_value = 2 * sp.stats.norm.cdf(-np.abs(z))
```

#### STEP #5

Compare $\text{p-value}$ with $\alpha$

```python
if p_value < alpha:
    print("Reject Null-Hypothesis i.e., accept Alternate-Hypothesis.")
else:
    print("Failed to reject Null-Hypothesis i.e., reject Alternate-Hypothesis.")
```

### 4.2 Hypothesis Testing using Confidence-Interval

#### STEP #1

Define Null-Hypothesis and Alternate-Hypothesis.

1. $H_0$ :
2. $H_a$ :

#### STEP #2

Identify:

1. Distribution:
2. Test Statistic: z-score
2. Significance level:

```python
alpha =   # Significance level.
```

#### STEP #3

Identify tail-type based on observed value.

```python
x =   # Observed value or Sample mean.
```

#### STEP #4

Compute Confidence Interval:

```python
import numpy as np
import scipy as sp


# Given,
mu =           # Population mean
sigma =        # Population standard deviation
n =            # Sample size
mu_sample =    # Sample mean

# Confidence level. For example: 0.99 (i.e., 99%)
confidence =  1 - alpha 

se = sigma / np.sqrt(n)  # Standard error

z_critical = sp.stats.norm.ppf(confidence)

moe = z_critical * se  # Margin of error

# Confidence Interval [x_1, x_2]
x_1 = mu_sample - moe
x_2 = mu_sample + moe
```

#### STEP #5

Compare population mean $\mu$ with Confidence Interval.

```python
if x_1 <= mu <= x_2:
    print("Failed to reject Null Hypothesis. Population mean falls in between the Confidence Interval.")
else:
    print("Reject Null Hypothesis. Population mean does not falls in between the Confidence Interval.")
```

## 5 Case study: Marketing Firms

### 5.1 Problem Statement

Suppose there is a Retail Store Chain that sells Shampoo bottles.  
This chain has **2000 stores** across India.  
The parameters for weekly sales of the shampoo bottle were reported as:
- Mean: 1800
- Standard deviation: 100

This was calculated by analyzing a lot of historical data.  
As a Manager / Owner / Data Scientist, you want to **increase these sales**, to generate more revenue.

**Q1. What are the techniques at your disposal?**

- Hire a marketing team

But there is an important factor to consider. These marketing teams/firms are not cheap, and would add a significant cost.  
It stands to reason that you would not straightaway hand over all 2000 stores to them.  
You would want an assurance that their work actually does impact the sales, and generate enough revenue that it is feasible to hire them.

**Q2. How would you get that assurance?**

Perhaps you can allot them a few stores, and analyze the sale parameters (Mean and Standard deviation).  
If results are good in a couple of weeks, then hire for all 2000 stores.

You decide to do this experiment with 2 competing marketing firms:

**Firm A**

- Worked on **50 stores**
- Sold an **average** **1850** bottles of shampoo

**Firm B**

- Worked on **5 stores**
- Sold an **average** **1900** bottles of shampoo

**Q1. Which firm gave better results?**

Clearly the sales are more for Firm B, but it seems that the number of stores under them were significantly less than Firm A.  
It is possible that this increase by Firm B is just a chance factor because the standard deviation of the population was 100.

**Q2. How do we quantify this and determine if it is just by chance or if it is actually statistically significant?**

When we talk about statistical significance, the word significance level pops into mind.  
Since this is a big decision that would affect revenue, you want to be very very sure (99% confidence) about your decision, i.e. $α=0.01$  
So, we need to employ the framework we saw and conduct hypothesis testing to see which firm's results are more significant.

### 5.2 Hypothesis Testing on Firm A

#### Hypothesis Testing using p-value

##### STEP #1

- $H_0 : \mu = 1800$
- $H_a : \mu > 1800$

##### STEP #2

Distribution: Normal Distribution  
Test Statistic: z-score  
Significance level: 0.01

In [2]:
alpha = 0.01

##### STEP #3

Since we have to find $P(x > 1850)$ we need to perform Right-Tailed test.

##### STEP #4

Compute $\text{p-value}$

In [3]:
mu = 1800
sigma = 100
n = 50

In [4]:
# Find: P(x > 1850)
# P(X > 1850) = 1 - P(X < 1850)
x = 1850

In [5]:
se = sigma / np.sqrt(n)  # Standard error
se.round(6).item()

14.142136

In [6]:
# x = mu + (z * sigma)
# For sample distribution sigma is Standard error.
z = (x - mu) / se
z.round(6).item()

3.535534

In [7]:
p_x_gt_1850 = 1 - sp.stats.norm.cdf(z)
p_value = p_x_gt_1850.round(6).item()
print("p-value:", p_value)

p-value: 0.000203


##### STEP #5

Compare $\text{p-value}$ with $\alpha$

In [8]:
if p_value <= alpha:
    print("Reject Null-Hypothesis i.e., accept Alternate-Hypothesis.")
else:
    print("Failed to reject Null-Hypothesis i.e., reject Alternate-Hypothesis.")

Reject Null-Hypothesis i.e., accept Alternate-Hypothesis.


Probability of finding data given that the null-hypothesis is true is very low hence we accept the alternate-hypothesis.

##### Critical Value

In [9]:
z_critical = sp.stats.norm.ppf(0.99)
z_critical.round(6).item()

2.326348

In [10]:
x_critical = mu + (z_critical * se)
np.floor(x_critical).item()

1832.0

#### Hypothesis Testing using CI

In [11]:
mu_sample = x

##### Compute Confidence Interval

In [12]:
moe = z_critical * se  # Margin of error
moe.round(6).item()

32.899527

In [13]:
x_1 = mu_sample - moe
x_2 = mu_sample + moe

print("Population mean:", mu)
x_1.round(6).item(), x_2.round(6).item()

Population mean: 1800


(1817.100473, 1882.899527)

In [14]:
if x_1 <= mu <= x_2:
    print("Failed to reject Null Hypothesis. Population mean falls in between the Confidence Interval.")
else:
    print("Reject Null Hypothesis. Population mean does not falls in between the Confidence Interval.")

Reject Null Hypothesis. Population mean does not falls in between the Confidence Interval.


#### Business Insights

1. Very low probability value of 0.0002 indicates that there is no enough evidence to backup the Null Hypothesis being true.
2. There is less than 1% probability i.e., around 0.02% chance that the average sales would have reached 1850 without the effort of Marketing Firm A.
3. There is a 99.98% probability that average sales improved from 1800 to 1850 after hiring Marketing Firm A.

### 5.3 Hypothesis Testing on Firm B

#### Hypothesis Testing using p-value

##### STEP #1

- $H_0 : \mu = 1800$
- $H_a : \mu > 1800$

##### STEP #2

Distribution: Normal Distribution  
Test Statistic: $\text{z-score}$  
Significance level: 0.01

In [15]:
alpha = 0.01

##### STEP #3

Since we have to find $P(x > 1850)$ we need to perform Right-Tailed test.

##### STEP #4

Compute $\text{p-value}$

In [16]:
mu = 1800
sigma = 100
n = 5

In [17]:
# Find: P(x > 1900)
x = 1900

In [18]:
se = sigma / np.sqrt(n)  # Standard error
se.round(6).item()

44.72136

In [19]:
# x = mu + (z * sigma)
# For sample distribution sigma is Standard error.
z = (x - mu) / se
z.round(6).item()

2.236068

In [20]:
p_x_gt_1900 = 1 - sp.stats.norm.cdf(z)
p_value = p_x_gt_1900.round(4).item()
print("p-value:", p_value)

p-value: 0.0127


##### STEP #5

Compare $\text{p-value}$ with $\alpha$

In [21]:
if p_value <= alpha:
    print("Reject Null-Hypothesis i.e., accept Alternate-Hypothesis.")
else:
    print("Failed to reject Null-Hypothesis i.e., reject Alternate-Hypothesis.")

Failed to reject Null-Hypothesis i.e., reject Alternate-Hypothesis.


In [22]:
0.0127 * 100, 100 - 0.0127 * 100

(1.27, 98.73)

There is a 1.27% chance of finding data given that the null-hypothesis is true, which is higher than the threshold value of 1%, hence we reject the alternate-hypothesis.

##### Critical Value

In [23]:
z_critical = sp.stats.norm.ppf(0.99)
z_critical.round(6).item()

2.326348

In [24]:
x_critical = z_critical * se + mu
np.floor(x_critical).item()

1904.0

#### Hypothesis Testing using CI

In [25]:
mu_sample = x

##### Compute Confidence Interval

In [26]:
moe = z_critical * se
moe.round(6).item()

104.03744

In [27]:
x_1 = mu_sample - moe
x_2 = mu_sample + moe

print("Population mean:", mu)
x_1.round(6).item(), x_2.round(6).item()

Population mean: 1800


(1795.96256, 2004.03744)

In [28]:
if x_1 <= mu <= x_2:
    print("Failed to reject Null Hypothesis. Population mean falls in between the Confidence Interval.")
else:
    print("Reject Null Hypothesis. Population mean does not falls in between the Confidence Interval.")

Failed to reject Null Hypothesis. Population mean falls in between the Confidence Interval.


#### Business Insights

1. The probability value of 0.0127 indicates that there is enough evidence to backup the Null Hypothesis being true.
2. There is around 1.27% chances that the average sales would have reached 1900 without the effort of Marketing Firm B.
3. There is around 98.73% chances that average sales improved from 1800 to 1900 after hiring Marketing Firm B.
4. Since we wanted to be 99% sure about the effectiveness of hiring Firm B, the confidence level is not met.
5. If Firm B had done an average sale of 1904 then they would have achieved 99% confidence level.

### 5.4 Conclusion

1. Firm B has 98.73% chances of improving the average sales from 1800 to 1900.
2. Firm A has 99.98% chances of improving the average sales from 1800 to 1850.
3. Because the confidence level of 98.73% is slightly below our 99% target for confirming a marketing  
   firm's effectiveness in improving average sales, we will proceed with Firm A rather than Firm B.

## 6 Quizzes

### 6.1 Quiz #1

A french cake shop claims that the average number of pastries they can produce in a day exceeds 500.  
The average number of pastries produced per day over a 70 day period was found to be 530.  
Assume that the population standard deviation for the pastries produced per day is 125.  

1. Test the claim using a z-test at the alpha (significance level) = 0.05, and state your interpretation.
2. Test the claim using a z-test with the critical z-value = 1.64

#### Solution

##### STEP #1

Define Null-Hypothesis and Alternate-Hypothesis.

- $H_0 : \mu = 500$
- $H_a : \mu > 500$

##### STEP #2

Identify:

1. Distribution: Normal Distribution
2. Test Statistic: z-score
3. Significance level: 0.05

In [29]:
alpha = 0.05

##### STEP #3

Identify tail-type.

Since we have to find $P(x > 530)$ we need to perform Right-Tailed test

In [30]:
# Observed value.
x = 530

##### STEP #4

Compute $\text{p-value}$

In [31]:
mu = 500
sigma = 125
n = 70

# Find: P(x > 530)
x = 530

In [32]:
se = sigma / np.sqrt(n)  # Standard error
se.round(6).item()

14.940358

In [33]:
# x = mu + (z * sigma)
# For sample distribution sigma is Standard error.
z = (x - mu) / se
z.round(6).item()

2.007984

In [34]:
p_x_gt_530 = 1 - sp.stats.norm.cdf(z)
p_value = p_x_gt_530.round(6).item()

print("p-value:", p_value)

p-value: 0.022322


##### STEP #5

Compare $\text{p-value}$ with $\alpha$

In [35]:
if p_value <= alpha:
    print("Reject Null-Hypothesis i.e., accept Alternate-Hypothesis.")
else:
    print("Failed to reject Null-Hypothesis i.e., reject Alternate-Hypothesis.")

Reject Null-Hypothesis i.e., accept Alternate-Hypothesis.


In [36]:
1 - p_value

0.977678

##### Interpretation

The claim that the average number of pastries they can produce in a day exceeds 500 is valid with confidence level of 97%.

In [37]:
z_critical = 1.64  # Given in the question.

p_x_gt_530 = 1 - sp.stats.norm.cdf(z_critical)
p_value = p_x_gt_530.round(6).item()

print("p-value:", p_value)

p-value: 0.050503


In [38]:
if p_value <= alpha:
    print("Reject Null-Hypothesis i.e., accept Alternate-Hypothesis.")
else:
    print("Failed to reject Null-Hypothesis i.e., reject Alternate-Hypothesis.")

Failed to reject Null-Hypothesis i.e., reject Alternate-Hypothesis.


In [39]:
1 - p_value

0.949497

##### Interpretation

The claim that the average number of pastries they can produce in a day exceeds 500 is valid.

##### Critical value

In [40]:
z_critical = sp.stats.norm.ppf(0.95)
z_critical.round(6).item()

1.644854

In [41]:
x_critical = mu + (z_critical * se)
np.floor(x_critical).item()

524.0

### 6.2 Quiz #2

A fitness App claims that its users walk an average of 8,000 steps per day.  
A random sample of 30 users showed an average of 7,600 steps per day with a standard deviation of 1,200 steps.  
Conduct a left-tailed Z-test at a 5% significance level to determine if the App's claim is supported. 

What is the p-value?

A. $\text{p-value $\approx$ 0.0339}$, Reject the Null Hypothesis.  
B. $\text{p-value $\approx$ 0.0762}$, Reject the Null Hypothesis.  
C. $\text{p-value $\approx$ 0.966}$, Fail to Reject the Null Hypothesis.  
D. $\text{p-value $\approx$ 0.3681}$, Fail to Reject the Null Hypothesis.  

#### Solution

##### STEP 1

- $H_0 : \mu = 8000$
- $H_a : \mu < 8000$

##### STEP 2

Identify:

1. Distribution: Normal Distribution
1. test-statistic: Z-Score
1. Significance level $(\alpha)$: 0.05

In [42]:
alpha = 0.05

##### STEP 3

Identify tailed-test type:

Given observed value of mean, sample mean 7600, is less than the population mean 8000.  
We have to find P(X < 7600), hence we need to perform left tailed test.

In [43]:
# Observed value.
x = 7600

##### STEP 4

Compute $\text{p-value}$

In [44]:
mu = 8000
sigma = 1200
n = 30

# Find: P(X < 7600)
x = 7600

In [45]:
se = sigma / np.sqrt(n)  # Standard error
se.round(6).item()

219.089023

In [46]:
# x = mu + (z * sigma)
# For sample distribution sigma is Standard error.
z_score = (x - mu) / se
z_score.round(6).item()

-1.825742

In [47]:
p_x_lt_7600 = sp.stats.norm.cdf(z_score)
p_value = p_x_lt_7600.round(6).item()
p_value

0.033945

##### STEP 5

Compare $\text{p-value}$ with $\alpha$

In [48]:
if p_value <= alpha:
    print("Reject Null-Hypothesis i.e., accept Alternate-Hypothesis.")
else:
    print("Failed to reject Null-Hypothesis i.e., reject Alternate-Hypothesis.")

Reject Null-Hypothesis i.e., accept Alternate-Hypothesis.


Answer:

A. $\text{p-value $\approx$ 0.0339}$, Reject the Null Hypothesis.

### 6.3 Quiz #3

In a dataset of exam scores with a mean of 60 and a standard deviation of 15,  
What is the critical value for the corresponding Z-score at a 95% confidence level?

#### Solution

In [49]:
mu = 60
sigma = 15

##### Left-tailed

In [50]:
z_score = sp.stats.norm.ppf(0.025)
z_score.round(6).item()

-1.959964

In [51]:
x_critical = mu + (z_score * sigma)
x_critical.round(6).item()

30.60054

##### Right-tailed

In [52]:
z_score = sp.stats.norm.ppf(0.95)
z_score.round(6).item()

1.644854

In [53]:
x_critical = mu + (z_score * sigma)
x_critical.round(6).item()

84.672804

##### Two-tailed

In [54]:
z1 = sp.stats.norm.ppf(0.025)
z2 = sp.stats.norm.ppf(1 - 0.025)
z1.round(6).item(), z2.round(6).item()

(-1.959964, 1.959964)

In [55]:
x1 = mu + (z1 * sigma)
x2 = mu + (z2 * sigma)
x1.round(6).item(), x2.round(6).item()

(30.60054, 89.39946)